gms/factories/src/main/java/com/linkedin/gms/factory/common/ElasticsearchSSLContextFactory.java

```diff
@@ -0,0 +1,85 @@
+package com.linkedin.gms.factory.common;
+
+import org.apache.http.ssl.SSLContextBuilder;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import javax.annotation.Nonnull;
+import javax.net.ssl.SSLContext;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.KeyManagementException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.cert.CertificateException;
+
+
+@Configuration
+public class ElasticsearchSSLContextFactory {
+
+    @Value("${ELASTICSEARCH_SSL_PROTOCOL:#{null}}")
```

Comment:
Should we provide reasonable defaults for some of these?

Reply:
Not sure. Actual behaviour when not specifying them is fallback to the java platform defaults. Maybe that's good enough?